testsuite/tests/lib-dynlink-csharp/Makefile

```diff
@@ -30,7 +30,9 @@ all: prepare bytecode bytecode-dll native native-dll
 prepare:
 	@if $(SUPPORTS_SHARED_LIBRARIES); then \
```

Comment:
Fixing the CSC check should fix the build because it won't build either bytecode or native if you don't have Microsoft's C# compiler. Your proposed patch is incomplete if you want to do it as a bytecode-only tests fix - there are two other places where those checks need adding (however, I have a vague recollection that the testsuite doesn't work for bytecode-only on Windows, as it happens)